lib/buffer.js

```diff
@@ -137,8 +137,14 @@ Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
 function assertSize(size) {
-  if (typeof size !== 'number') {
-    const err = new TypeError('"size" argument must be a number');
+  let err = null;
+
+  if (typeof size !== 'number')
+    err = new TypeError('"size" argument must be a number');
+  if (size < 0)
```

Comment:
Perhaps `else if` here so that it wouldn't attempt to call `.valueOf` etc. on objects?